### DISCUSSION TITLE
Search for an Image within a bigger Image

### DISCUSSION THREAD

**Discussion by tomasmueller** [ImageMagick/ImageMagick#2969] [Help]
Hi,

i want to search for an image that is a part of a bigger image. This should be possible with 'compare' and 'subimage' but i could not find out how it works. I create a full screen screenshot and want to search for a windows that is placed somewhere in the screenshot. For some reasons i cannot get the position with other tools (wmctrl on linux).

I have a screenshot of the window and create a new screenshot of the while screen. Now IM should search for the small image within the bigger image and give me the position of the subimage.

is this possible?

**Comment by snibgo** [comment]
For example, we search for a 2x2 white image within the built-in "rose:" image, using RMSE as the measure:
```
magick compare -metric RMSE -subimage-search rose: -size 2x2 xc:White NULL:

74.1895 (0.00113206) @ 62,15
```
The number in parentheses is not zero, so no exact match was found. But the number is very low (less than 0.01) so the closest match was very close. The top-left pixel of the subimage was at (62,15) in the main image.

**Comment by fmw42** [comment]
See 

https://imagemagick.org/script/compare.php
https://imagemagick.org/Usage/compare/

https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=14613&p=51076&hilit=compare+metric#p51076

https://www.imagemagick.org/discourse-server/viewtopic.php?p=51093&sid=4ed368a47222187accb5380a2e4fbafd#p51093
